src/main.rs

```diff
@@ -1,18 +1,37 @@
+use std::env;
+use std::process::exit;
+
+use clap::{crate_authors, Clap};
+
 mod rusty_hook;
 
-use std::process::exit;
+#[derive(Clap)]
+#[clap(version = env!("CARGO_PKG_VERSION"), author = crate_authors!())]
+struct RustyHookOpts {
+    #[clap(subcommand)]
+    subcmd: RustyHookSubCommand,
+}
 
-extern crate ci_info;
-extern crate getopts;
-extern crate nias;
-use getopts::Options;
-use std::env;
+#[derive(Clap)]
+enum RustyHookSubCommand {
+    /// Initialize rusty-hook's git hooks in the current directory.
+    #[clap(version = env!("CARGO_PKG_VERSION"), author = crate_authors!())]
+    Init,
+    /// Run a git hook using the current directory's configuration.
+    /// Ran automatically by rusty-hook's git hooks.
+    #[clap(version = env!("CARGO_PKG_VERSION"), author = crate_authors!())]
+    Run(RustyHookRun),
+}
 
-fn print_version() {
-    println!(env!("CARGO_PKG_VERSION"));
+#[derive(Clap)]
+struct RustyHookRun {
+    #[clap(long)]
+    hook: String,
+    #[clap(name = "git args", raw(true))]
+    args: Vec<String>,
```

Comment:
Getting a runtime error with these updates

```bash
$ git commit -m "foo"
error: Found argument '-v' which wasn't expected, or isn't valid in this context

If you tried to supply `-v` as a PATTERN use `-- -v`

USAGE:
    rusty-hook <SUBCOMMAND>

For more information try --help
error: The argument '<git args>...' requires a value but none was supplied

USAGE:
    rusty-hook run --hook <hook> [-- <git args>...]

For more information try --help
Configured hook command failed
pre-commit hook rejected
```

Reply:
Ah I think this is because the shorthand `-v` is no longer available and clap requires `-V`

Reply:
The `-v` bit can be fixed [here](https://github.com/swellaby/rusty-hook/blob/master/src/hook_files/cli.sh#L29), but looks like some additional adjustments are needed here. Does clap need this to be Option wrapped?

```bash
$ git commit -m "foo"
error: The argument '<git args>...' requires a value but none was supplied

USAGE:
    rusty-hook run --hook <hook> [-- <git args>...]

For more information try --help
Configured hook command failed
pre-commit hook rejected
```

Reply:
Should be fixed.

Reply:
This is still failing unfortunately. I believe what's happening with some hooks that don't have any additional args is that the hook scripts end up invoking something like:

```bash
$ rusty-hook run --hook pre-push ""
```

and clap doesn't like the empty string.

I'm not sure if there's a way to coerce clap into accepting an empty string, but I guess we could work around it on the script side regardless.

https://github.com/swellaby/rusty-hook/blob/master/src/hook_files/hook_script.sh#L24

Could either conditionally tack on the arg only if there actually were get params, or even just sticking a space in the arg (`" ${gitParams}"`) so that clap will always accept it, and then trim within rusty-hook

Reply:
Erm, latest commit passed CI?

Reply:
The CI checks run the unit tests which did indeed pass, but there aren't any automated functional test so there's nothing in CI actually running the rusty-hook cli and verifying behavior. You can test this locally and reproduce yourself via `$ cargo run --bin rusty-hook -- run --hook "pre-commit" -- ""` or installing your updated version `cargo install --path . --force` (you may need to `cargo clean && cargo build to update the local hooks) and then trying to commit

Reply:
I've fixed it in the script:
```sh
rusty-hook run --hook "${hookName}" $([ -z "$gitParams" ] && echo "-- $gitParams" || echo "")
```